bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -93,6 +96,32 @@ protected void modified(@Nullable Map<String, @Nullable Object> properties) {
         }
     }
 
+    private SecurityContext authenticateBearerToken(String token) throws AuthenticationException {
+        if (token.startsWith(API_TOKEN_PREFIX)) {
+            UserApiTokenCredentials credentials = new UserApiTokenCredentials(token);
+            Authentication auth = userRegistry.authenticate(credentials);
+            User user = userRegistry.get(auth.getUsername());
+            if (user == null) {
+                throw new org.openhab.core.auth.AuthenticationException("User not found in registry");
```

Comment:
```suggestion
                throw new AuthenticationException("User not found in registry");
```

Reply:
Thanks, I rebased and addressed it in the process.